### ISSUE TITLE
[Feature] Serialize specific tests in an otherwise-parallelized test suite

### ISSUE THREAD

**Issue by erikthiem-immuta** [microsoft/playwright#11655]
Hi there! I am writing a suite of Playwright tests. So far it has mostly been a joy to use.

However, I have run into some issues when parallelizing tests. I am pretty sure that this functionality does not exist, but if it does, happy to learn more about it.

The vast majority of my tests are independent and can be run in parallel without any issues. Test objects are created at the start of the test and cleaned up at the end. No problem. However, I have a few tests that necessarily affect global state in the application. These can still clean themselves up at the end so the state at the end of the test is the same as the start. However, they can't be run in parallel with any other test because their effects while running will mess with other tests.

So far I have solved this by only using one worker and running no tests in parallel. This is not ideal, as it slows tests down a bit. **Is there (or can there be) some way to run all tests with the specified number of workers but run x tests (demarcated somehow in the test) only by themselves?**

I looked into https://playwright.dev/docs/api/class-test#test-describe-serial and thought I had found a way to do this, but this appears to only run tests in a given `describe` block in serial and not be what I'm actually looking for.

My next attempt is probably going to be to try to use some sort of semaphore to get certain tests only to run on their own, but this will be kinda hacky and it would be nice if the framework natively supported this feature.

Thank you!

**Comment by aslushnikov** [comment]
@erikthiem-immuta The way I'd do this is by creating 2 separate projects:
- one project with all your independent parallel tests
- second project with all your tests that must be run serially.

After that, I'd run these tests separately using CLI:

```bash
$ npx playwright test --project fast-indepent-tests
$ npx playwright test --project slow-serial-tests --workers 1
```

**Comment by erikthiem-immuta** [comment]
Thanks, @aslushnikov! I had considered doing it this way, but was worried that this would ultimately end up being slower due to duplicating the (rather complex and slow) test setup that I have running before the tests started. Still, I'll give it a shot. As the number of tests increase and the proportion of tests spent running vs. setting up increases, this two project approach would end up being faster than using a single worker.

**Comment by aslushnikov** [comment]
K! Let me close this then; feel free to file another issue if / when problems happen!

**Comment by amitz** [comment]
Same issue here. The two main issues we encountered:

- You're getting two reports, and you need to combine them into a single one
- our global setup/teardown are lengthy. They create multiple entities in the system and clean them all when the run ends. This basically double the run time.

A semi workaround, would be implementing [this](https://github.com/microsoft/playwright/issues/21229). It will allow two projects to share the same global-setup and tear-down (and as they running on the same process, output the same Report file).
